lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -0,0 +1,257 @@
+#include "SwiftUnsafeTypes.h"
+
+#include "lldb/DataFormatters/TypeSynthetic.h"
+#include "lldb/Symbol/SwiftASTContext.h"
+#include "lldb/Target/SwiftLanguageRuntime.h"
+
+#include <utility>
+
+using namespace lldb;
+using namespace lldb_private;
+
+class SwiftUnsafeBufferPointer {
+public:
+  SwiftUnsafeBufferPointer(ValueObject &valobj);
+  size_t GetCount() const { return m_count; }
+  addr_t GetStartAddress() const { return m_start_addr; }
+  CompilerType GetElementType() const { return m_elem_type; }
+  bool IsValid() const;
+  bool Update();
+
+private:
+  ValueObject &m_valobj;
+  size_t m_count;
+  addr_t m_start_addr;
+  CompilerType m_elem_type;
+};
+
+SwiftUnsafeBufferPointer::SwiftUnsafeBufferPointer(ValueObject &valobj)
+    : m_valobj(*valobj.GetNonSyntheticValue().get()) {}
+
+bool SwiftUnsafeBufferPointer::IsValid() const {
+  return m_count != UINT64_MAX && m_start_addr != LLDB_INVALID_ADDRESS &&
+         m_start_addr != 0;
+}
+
+bool SwiftUnsafeBufferPointer::Update() {
+  if (!m_valobj.GetNumChildren())
+    return false;
+
+  static ConstString g_count("count");
+  ValueObjectSP value_sp(m_valobj.GetChildMemberWithName(g_count, true));
+  if (!value_sp)
+    return false;
+
+  ValueObjectSP child_valobj_sp = nullptr;
+  ValueObjectSP synthetic = value_sp->GetSyntheticValue();
+  if (synthetic)
+    child_valobj_sp = synthetic->GetChildAtIndex(0, true);
+  if (!child_valobj_sp)
+    child_valobj_sp = value_sp->GetChildAtIndex(0, true);
+  if (!child_valobj_sp)
+    return false;
```

Comment:
I think this logic deserves comments to explain which cases will lead to which control flow and corresponding testing coverage.